proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/sdk/PowerAuthSDK.java

```diff
@@ -1599,4 +1599,46 @@ public void onFingerprintInfoDialogClosed() {
         dialog.show();
     }
 
+    // E2EE
+
+    /**
+     * Creates a new instance of ECIES encryptor suited for application's general end-to-end encryption purposes.
+     * The returned encryptor is cryptographically bounded to the PowerAuth configuration, so it can be used
+     * with or without the valid activation.
+     *
+     * @return New instance of {@link EciesEncryptor} object or {@code null} if {@link PowerAuthConfiguration}
+     *         contains an invalid data.
+     */
+    public @Nullable EciesEncryptor getEciesEncryptorForApplicationScope() {
+        try {
+            final IPrivateCryptoHelper helper = getCryptoHelper(null);
+            return helper.getEciesEncryptor(EciesEncryptorId.GENERIC_APPLICATION_SCOPE);
+        } catch (PowerAuthErrorException e) {
+            PA2Log.e("Failed to create application scoped ECIES encryptor: " + e.getMessage());
+            return null;
```

Comment:
Or maybe it should throw an InvalidStateException instead of returning null (in both ecies methods).

Reply:
Right. I thought that we don't throw an exceptions from that object (because the class was rewritten from objc), but I was wrong. For this case it's OK to throw that `PowerAuthErrorException`